Sources/SwiftDocC/Utility/ValidatedURL.swift

```diff
@@ -55,10 +55,20 @@ public struct ValidatedURL: Hashable, Equatable {
     ///
     /// Use this to parse author provided documentation links that may contain links to on-page subsections. Escaping the fragment allows authors
     /// to write links to subsections using characters that wouldn't otherwise be allowed in a fragment of a URL.
+    ///
+    /// - Important: Documentation links don't include query items but "?" may appear in the link's path.
     init?(parsingAuthoredLink string: String) {
         // Try to parse the string without escaping anything
-        if let parsed = ValidatedURL(parsingExact: string) {
-            self.components = parsed.components
+        if var parsedComponents = ValidatedURL(parsingExact: string)?.components {
+            // Documentation links don't include query items but "?" may appear in the link's path.
+            // If `URLComponents` decoded a `query`, move that over to the `path` instead.
```

Comment:
I think this is slightly counter-intuitive and could benefit from an explanation as to how moving to `path` benefits us down the line

Reply:
How about this comment?
```suggestion
            // If `URLComponents` decoded a `query`, that's correct from a general URL standpoint but incorrect from a documentation link standpoint.
            // To create a valid documentation links, we move the `query` component and its "?" separator into the `path` component.
```

Reply:
Gotcha yes that makes sense, thank you!

Reply:
Updated in https://github.com/swiftlang/swift-docc/pull/1159/commits/7e360941b9f1d58e4a76513a6bd05e449554ede6